aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecorator.kt

```diff
@@ -61,7 +62,9 @@ class SigV4AuthDecorator : ConditionalDecorator(
     delegateTo =
         object : ClientCodegenDecorator {
             override val name: String get() = "SigV4AuthDecorator"
-            override val order: Byte = 0
+
+            // This decorator must decorate before S3ExpressDecorator so that sigv4 appears before sigv4-s3express within auth_scheme_options
+            override val order: Byte = (S3ExpressDecorator.ORDER + 1).toByte()
```

Comment:
It's a little strange for the generic sigv4 decorator to reference a specific S3 customization. Should be the other way around. Make the S3 decorator reference this one's order.

Reply:
I did the other around for the first time I implemented but I switched it to be the way it is, because companion object cannot be placed into a local class. But it didn't look as bad, so I will revert it to match your suggestion.

Reply:
Addressed in 726926d & 21f62c3